src/init/VersionInitializer.ts

```diff
@@ -0,0 +1,29 @@
+import { readJson } from 'fs-extra';
+import type { KeyValueStorage } from '../storage/keyvalue/KeyValueStorage';
+import { modulePathPlaceholder, resolveAssetPath } from '../util/PathUtil';
+import { Initializer } from './Initializer';
+
+const STORAGE_KEY = 'current-server-version';
+const PACKAGE_JSON_PATH = `${modulePathPlaceholder}package.json`;
+
+/**
+ * This initializer simply writes the version number of the server to the storage.
+ * This will be relevant in the future when we look into migration initializers.
+ *
+ * It automatically parses the version number from the `package.json`.
+ */
+export class VersionInitializer extends Initializer {
+  private readonly storage: KeyValueStorage<string, string>;
+
+  public constructor(storage: KeyValueStorage<string, string>) {
+    super();
+    this.storage = storage;
+  }
+
+  public async handle(): Promise<void> {
+    const path = resolveAssetPath(PACKAGE_JSON_PATH);
+    const pkg = await readJson(path);
```

Comment:
I mostly did it because we already had the import anyway (and I only noticed it was a dev dependency instead of a full dependency after the changes 😅). But yea, it's not really needed so I could revert if we care about the extra size of the installed package.

Reply:
I didn't check, but I do assume it could probably be used in the `FileDataAccessor` for certain scenarios.

Reply:
Good be good first-issue.